paymentsheet/src/main/java/com/stripe/android/paymentsheet/IntentConfirmationHandler.kt

```diff
@@ -82,18 +86,21 @@ internal class IntentConfirmationHandler(
      */
     val hasReloadedFromProcessDeath = hasReloadedWhileAwaitingPreConfirm || hasReloadedWhileAwaitingConfirm
 
-    private var completableResult: CompletableDeferred<Result>? = if (hasReloadedFromProcessDeath) {
-        CompletableDeferred()
-    } else {
-        null
-    }
+    private val _state = MutableStateFlow(
+        if (hasReloadedFromProcessDeath) {
+            State.Confirming
+        } else {
+            State.Idle
+        }
+    )
+    val state: StateFlow<State> = _state.asStateFlow()
 
     init {
         if (hasReloadedWhileAwaitingConfirm) {
             coroutineScope.launch {
                 delay(1.seconds)
```

Comment:
is this still needed?

Reply:
Yup! We still want to cancel payment confirmation if we don't receive a result from an payment confirmation activity (3DS, Cash App, etc...) for some reason and not be stuck in a confirming state.